fms/include/fms.inc

```diff
@@ -16,729 +16,25 @@
-!! @details Tries to find a match for a character string in a list of character strings.
-!!      The match is case sensitive and disregards blank characters to the right of
-!!      the string.
-!!
-!!      <br>Examples<br>
-!!      @code{.F90}
-!!       string = "def"
-!!       string_array = (/ "abcd", "def ", "fghi" /)
-!!
-!!       string_array_index ( string, string_array, index )
-!!      @endcode
-!!       Returns: TRUE, index = 2
-function string_array_index ( string, string_array, index ) result (found)
-character(len=*),  intent(in)  :: string !< Character string of arbitrary length.
-character(len=*),  intent(in)  :: string_array(:) !< Array/list of character strings.
-integer, optional, intent(out) :: index !< The index of string_array where the first match was found. If
-                                        !! no match was found then index = 0.
-logical :: found !< If an exact match was found then TRUE is returned, otherwise FALSE is returned.
-integer :: i
-
-! initialize this function to false
-! loop thru string_array and exit when a match is found
-
-  found = .false.
-  if (present(index)) index = 0
-
-  do i = 1, size(string_array(:))
-    ! found a string match ?
-    if ( trim(string) == trim(string_array(i)) ) then
-         found = .true.
-         if (present(index)) index = i
-         exit
-    endif
-  enddo
-
-end function string_array_index
-
-!#######################################################################
 
 !> @brief Determines if a real input array has monotonically increasing or
 !!     decreasing values.
 !! @return If the input array of real values either increases or decreases monotonically then true
 !! is returned, otherwise false is returned.
-function monotonic_array ( array, direction )
-real,    intent(in)            :: array(:) !< An array of real values. If the size(array) < 2 this function
-                                           !! assumes the array is not monotonic, no fatal error will occur.
+
+function MONOTONIC_ARRAY_(array, direction)
+real(FMS_MOD_KIND_), intent(in) :: array(:) !< An array of real values. If the size(array) < 2 this function
+                                            !! assumes the array is not monotonic, no fatal error will occur.
 integer, intent(out), optional :: direction !< If the input array is:
                                             !! >> monotonic (small to large) then direction = +1.
                                             !! >> monotonic (large to small) then direction = -1.
                                             !! >> not monotonic then direction = 0.
-logical :: monotonic_array !< If the input array of real values either increases or decreases monotonically
-                           !! then TRUE is returned, otherwise FALSE is returned.
+logical :: MONOTONIC_ARRAY_ !< If the input array of real values either increases or decreases monotonically
```

Comment:
I realize you didn't write this code @J-Lentz , but this seems very odd to me.  I would just declare this function as logical, or have a return value and declare that return value as logical.  This is somehow both and neither at the same time.

Reply:
Return variable has been replaced with `ret` in c245ad67f1778eca3a7543cd82cdb57db54a426e.